V0.1 Alpha

FabrizioMoggio opened the pull request:
#### What type of PR is this?
* documentation

#### What this PR does / why we need it:
alpha release

#### Which issue(s) this PR fixes:
#36

FabrizioMoggio committed:
Update API version to 0.1.0 in YAML file

https://github.com/camaraproject/IoTNetworkOptimization/issues/36

FabrizioMoggio committed:
readme v0.1

https://github.com/camaraproject/IoTNetworkOptimization/issues/36

FabrizioMoggio committed:
alpha

FabrizioMoggio committed:
Update API readiness checklist with status notes

https://github.com/camaraproject/IoTNetworkOptimization/issues/36

FabrizioMoggio committed:
Merge branch 'camaraproject:main' into v0.1

FabrizioMoggio committed:
alpha release

https://github.com/camaraproject/IoTNetworkOptimization/issues/36

FabrizioMoggio committed:
Fix version number

FabrizioMoggio committed:
Changelog

FabrizioMoggio committed:
Update API versioning convention to alpha

FabrizioMoggio committed:
Update release information to reflect new version

FabrizioMoggio commented:
@camaraproject/release-management_reviewers this version is not intended to be part of the Meta Release.

varanto reviewed (dismissed):
ok for me

FabrizioMoggio committed:
Increase maxItems for event types from 1 to 2

Kevsy commented on code/API_definitions/iot-network-optimization.yaml:
```diff
@@ -128,7 +128,7 @@ externalDocs:
 ############################################################################
 servers:
-  - url: "{apiRoot}/iot-network-opt/vwip"
+  - url: "{apiRoot}/iot-network-opt/v0.1alpha1"
```
```suggestion
  - url: "{apiRoot}/iot-network-optimization/v0.1alpha1"
```

Kevsy commented on documentation/API_documentation/iot-network-optimization-API-Readiness-Checklist.md:
```diff
@@ -5,16 +5,16 @@ Checklist for kyc-match vwip
+| 10 | API release numbering convention applied     |   M   |         M         |    M    |    M   |  Y   |   r1.1   |
+| 11 | Change log updated                           |   M   |         M         |    M    |    M   |  Y   |      |
 | 12 | Previous public release was certified        |   O   |         O         |    O    |    M   |  N   |      |
 | 13 | API description (for marketing)              |   O   |         O         |    M    |    M   |      | [wiki link](https://lf-camaraproject.atlassian.net/wiki/spaces/CAM/pages/129204225/IoTNetworkOptimization) |
```
```suggestion
| 13 | API description (for marketing)              |   O   |         O         |    M    |    M   |      | [wiki link](https://lf-camaraproject.atlassian.net/wiki/x/AQCGDg) |
```

Kevsy replied:
^ small change to use the short 'share' link URL from the wiki

Kevsy reviewed (changes requested):
Just needs some small changes (see inline changes above) - otherwise LGTM

FabrizioMoggio committed:
409-202

•	The success response is now 202 Accepted instead of 200 OK, to reflect that the activation is processed asynchronously.
•	A 409 Conflict response has been added for cases where one or more devices in the request already have an active power-saving configuration.

FabrizioMoggio committed:
Update code/API_definitions/iot-network-optimization.yaml

Co-authored-by: Kevin Smith <Kevsy@users.noreply.github.com>

FabrizioMoggio committed:
Update documentation/API_documentation/iot-network-optimization-API-Readiness-Checklist.md

Co-authored-by: Kevin Smith <Kevsy@users.noreply.github.com>

Kevsy reviewed (approved):
LGTM
